modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/service/test/NotificationQueueEntryLocalServiceTest.java

```diff
@@ -133,66 +128,18 @@ public void testResendNotificationQueueEntry() throws Exception {
 	private NotificationQueueEntry _addNotificationQueueEntry()
 		throws Exception {
 
-		return _notificationQueueEntryLocalService.addNotificationQueueEntry(
-			_createNotificationContext(
-				TestPropsValues.getUser(), StringUtil.randomString(),
-				StringUtil.randomString(), NotificationConstants.TYPE_EMAIL));
+		return _addNotificationQueueEntry(
+			TestPropsValues.getUser(), StringUtil.randomString(),
+			StringUtil.randomString(), NotificationConstants.TYPE_EMAIL);
 	}
 
 	private NotificationQueueEntry _addNotificationQueueEntry(
 			User user, String body, String subject, String type)
 		throws Exception {
 
 		return _notificationQueueEntryLocalService.addNotificationQueueEntry(
-			_createNotificationContext(user, body, subject, type));
-	}
-
-	private NotificationContext _createNotificationContext(
-			User user, String body, String subject, String type)
-		throws Exception {
-
-		NotificationContext notificationContext = new NotificationContext();
-
-		NotificationQueueEntry notificationQueueEntry =
-			_notificationQueueEntryLocalService.createNotificationQueueEntry(
-				RandomTestUtil.randomInt());
-
-		notificationQueueEntry.setUserId(user.getUserId());
-		notificationQueueEntry.setUserName(user.getFullName());
-		notificationQueueEntry.setBody(body);
-		notificationQueueEntry.setSubject(subject);
-		notificationQueueEntry.setType(type);
-		notificationQueueEntry.setStatus(
-			NotificationQueueEntryConstants.STATUS_UNSENT);
-
-		notificationContext.setNotificationQueueEntry(notificationQueueEntry);
-
-		NotificationRecipient notificationRecipient =
-			_notificationRecipientLocalService.createNotificationRecipient(
-				_counterLocalService.increment());
-
-		notificationRecipient.setClassName(
-			NotificationQueueEntry.class.getName());
-		notificationRecipient.setClassPK(
-			notificationQueueEntry.getNotificationQueueEntryId());
-
-		notificationContext.setNotificationRecipient(notificationRecipient);
-
-		NotificationRecipientSetting notificationRecipientSetting =
-			_notificationRecipientSettingLocalService.
-				createNotificationRecipientSetting(
-					_counterLocalService.increment());
-
-		notificationRecipientSetting.setNotificationRecipientId(
-			notificationRecipient.getNotificationRecipientId());
-
-		notificationContext.setNotificationRecipientSettings(
-			Arrays.asList(notificationRecipientSetting));
-
-		notificationContext.setType(
-			NotificationConstants.TYPE_USER_NOTIFICATION);
-
-		return notificationContext;
+			NotificationTemplateUtil.createNotificationContext(
+				user, body, null, subject, type));
 	}
 
 	@Inject
```

Comment:
Since you refactored this class, I believe these variables are not being used anymore: _counterLocalService, _notificationRecipientLocalService, _notificationRecipientSettingLocalService, _notificationTemplateLocalService;

Could you please double-check? If positive, could you please remove it?